filters/openpolicyagent/openpolicyagent.go

```diff
@@ -476,38 +477,77 @@ func (registry *OpenPolicyAgentRegistry) new(store storage.Store, configBytes []
 // Start asynchronously starts the policy engine's plugins that download
 // policies, report status, etc.
 func (opa *OpenPolicyAgentInstance) Start(ctx context.Context, timeout time.Duration) error {
-	err := opa.manager.Start(ctx)
 
-	if err != nil {
-		return err
-	}
+	discoveryPlugin := discovery.Lookup(opa.manager)
 
-	// check readiness of all plugins
-	pluginsReady := func() bool {
-		for _, status := range opa.manager.PluginStatus() {
-			if status != nil && status.State != plugins.StateOK {
-				return false
+	done := make(chan struct{})
+	failed := make(chan error, 1)
+
+	var registerDiscoveryListenerOnce sync.Once
+	registerDiscoveryListenerOnce.Do(func() {
+		discoveryPlugin.RegisterListener("startuplistener", func(status bundle.Status) {
+			handleStatusErrors(status, failed, "discovery plugin")
+		})
+	})
+
+	// Register listener for "bundle" status
+	var registerBundleListenerOnce sync.Once
+	opa.manager.RegisterPluginStatusListener("bundlelistener", func(status map[string]*plugins.Status) {
```

Comment:
This applies to all the other listeners as well: the names (first parameter) are global names in the opa instance, so might yield collisions. 
I‘d suggest sth more unique: `skipper-instance-startup`, also move it to an internal constant..